### ISSUE TITLE
Does the "similar" method pollute the Base enviroment?

### ISSUE THREAD

**Issue by FaresX** [JuliaIntervals/IntervalArithmetic.jl#692]
https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/a54373119438fb8af50c05e7a504e09c16038ef3/src/matmul.jl#L156
This case can run successfully
```julia
using Configurations
@option mutable struct A
    a::Vector{Int} = []
end
a = A()
da = to_dict(a)
da["a"] = Union{}[]
a2 = from_dict(A, da)
```
but this can not
```julia
using Configurations
using IntervalArithmetic
@option mutable struct A
    a::Vector{Int} = []
end
a = A()
da = to_dict(a)
da["a"] = Union{}[]
a2 = from_dict(A, da)
```
The error is as follows
```
ERROR: ArgumentError: cannot construct a value of type Union{} for return result
Stacktrace:
  [1] (::Core.TypeofBottom)(a::Int64)
    @ Core .\boot.jl:318
  [2] zero(::Type{Union{}})
    @ Base .\number.jl:310
  [3] zeros(::Type{Union{}}, dims::Tuple{Int64})
    @ Base .\array.jl:590
  [4] similar(a::Vector{Union{}}, S::Type, dims::Tuple{Int64})
    @ IntervalArithmetic C:\Users\22112\.julia\packages\IntervalArithmetic\jXJeJ\src\matmul.jl:156
  [5] similar(a::Vector{Union{}}, ::Type{Union{}}, dims::Tuple{Base.OneTo{Int64}})
    @ Base .\abstractarray.jl:830
  [6] _similar_for(c::Vector{…}, ::Type{…}, itr::Base.Generator{…}, ::Base.HasShape{…}, axs::Tuple{…})
    @ Base .\array.jl:670
  [7] _collect(c::Vector{…}, itr::Base.Generator{…}, ::Base.EltypeUnknown, isz::Base.HasShape{…})
    @ Base .\array.jl:813
  [8] collect_similar(cont::Vector{Union{}}, itr::Base.Generator{Vector{…}, Configurations.var"#8#10"{…}})
    @ Base .\array.jl:720
  [9] map(f::Function, A::Vector{Union{}})
    @ Base .\abstractarray.jl:3371
 [10] from_dict(::Type{A}, of::OptionField{:a}, ::Type{Vector{Int64}}, x::Vector{Union{}})
    @ Configurations C:\Users\22112\.julia\packages\Configurations\Yxczn\src\from_dict.jl:136
 [11] macro expansion
    @ C:\Users\22112\.julia\packages\Configurations\Yxczn\src\from_dict.jl:372 [inlined]
 [12] macro expansion
    @ C:\Users\22112\.julia\packages\Configurations\Yxczn\src\codegen.jl:0 [inlined]
 [13] from_dict_specialize(::Type{A}, d::OrderedCollections.OrderedDict{String, Any})
    @ Main C:\Users\22112\.julia\packages\Configurations\Yxczn\src\codegen.jl:362
 [14] #from_dict#4
    @ C:\Users\22112\.julia\packages\Configurations\Yxczn\src\from_dict.jl:46 [inlined]
 [15] from_dict(::Type{A}, d::OrderedCollections.OrderedDict{String, Any})
    @ Configurations C:\Users\22112\.julia\packages\Configurations\Yxczn\src\from_dict.jl:33
 [16] top-level scope
    @ c:\Users\22112\OneDrive - mails.ucas.ac.cn\文档\CODE\Julia\MyPackages\QInsControl-Makie\QInsControl.jl\test\runtests.jl:17
Some type information was truncated. Use `show(err)` to see complete types.
```

**Comment by OlivierHnt** [comment]
Thank you reporting this issue. Oh it seems that we inadvertently overwrote the behaviour for eg `similar(Union{}[])` when specializing on `RealOrComplexI{T}` (i.e., `Union{Complex{Interval{T}}, Interval{T}}`).

I will fix it right away.

**Comment by OlivierHnt** [comment]
Should be fixed in the release v0.22.21. Feel free to re-open this issue or open a new one if you still run into troubles. Thanks again, and happy holidays!

**Comment by OlivierHnt** [comment]
Should be fixed in the release v0.22.21. Feel free to re-open this issue or open a new one if you still run into troubles. Thanks again, and happy holidays!
